### ISSUE TITLE
Ask for analytics BEFORE connecting to Google analytics

### ISSUE THREAD

**Issue by martin-braun** [flutter/flutter#99066]
EU GDPR laws require to ask for permission, before starting analytics, yet `flutter` will attempt to connect to Google analytics without permission. Yes, we can disable that using `flutter config --no-analytics`, but the EU GDPR laws require an explicit opt-in instead of an opt-out by the user to process any user data. 

## Proposal

When running flutter for the first time, ask for permission. This should be able to be bypassed with a flag, so scripts can work without any blocker, otherwise it will prevent scripts that install and use flutter without manual intervention from working, but in any case this change will break such scripts, so this change would be made in the next major update. 

Simply notifying about the default analytics in the docs is not enough. Alternative would be to disable analytics by default and nag about it as a warning until a decision has been made. This way legacy support will be maintained and no major version is required to respect the privacy of your users.

**Comment by stuartmorgan** [comment]
Duplicate of https://github.com/flutter/flutter/issues/73657

**Comment by martin-braun** [comment]
@stuartmorgan missed that, thanks!

**Comment by martin-braun** [comment]
@stuartmorgan Thanks I missed that, but it got locked and I have no understanding why you would do so. Locking prevents people from upvoting. I don't see how a discussion would be "sensitive" as he said, please unlock the issue or at least give an update, thanks.
